frontend/src/metabase/actions/containers/CreateActionForm/CreateActionForm.tsx

```diff
@@ -0,0 +1,126 @@
+import Form from "metabase/core/components/Form";
+import FormFooter from "metabase/core/components/FormFooter";
+import FormProvider from "metabase/core/components/FormProvider";
+import FormInput from "metabase/core/components/FormInput";
+import FormTextArea from "metabase/core/components/FormTextArea";
+import FormSubmitButton from "metabase/core/components/FormSubmitButton";
+import FormErrorMessage from "metabase/core/components/FormErrorMessage";
+
+import * as Errors from "metabase/core/utils/errors";
+
+import Actions, { CreateQueryActionOptions } from "metabase/entities/actions";
+
+import FormModelPicker from "metabase/models/containers/FormModelPicker";
+
+import type {
+  ActionFormSettings,
+  CardId,
+  WritebackQueryAction,
+} from "metabase-types/api";
+import type { State } from "metabase-types/store";
+import type Question from "metabase-lib/Question";
+
+const ACTION_SCHEMA = Yup.object({
+  name: Yup.string()
+    .required(Errors.required)
+    .max(100, Errors.maxLength)
+    .default(""),
+  description: Yup.string().nullable().max(255, Errors.maxLength).default(null),
+  model_id: Yup.number().required(Errors.required),
+});
+
+type FormValues = Pick<
+  CreateQueryActionOptions,
+  "name" | "description" | "model_id"
+>;
+
+interface OwnProps {
+  question: Question;
+  formSettings: ActionFormSettings;
+  modelId?: CardId;
+  onCreate?: (values: WritebackQueryAction) => void;
+  onCancel?: () => void;
+}
+
+interface DispatchProps {
+  handleCreateAction: (
+    action: CreateQueryActionOptions,
+  ) => Promise<WritebackQueryAction>;
+}
+
+type Props = OwnProps & DispatchProps;
+
+const mapDispatchToProps = {
+  handleCreateAction: Actions.actions.create,
+};
+
+function CreateActionForm({
+  question,
+  formSettings,
+  modelId,
+  handleCreateAction,
+  onCreate,
+  onCancel,
+}: Props) {
+  const initialValues = useMemo(
+    () => ({ ...ACTION_SCHEMA.getDefault(), model_id: modelId }),
+    [modelId],
+  );
+
+  const handleCreate = useCallback(
+    async (values: FormValues) => {
+      const reduxAction = await handleCreateAction({
+        ...values,
+        question,
+        formSettings,
+      });
+      const action = Actions.HACK_getObjectFromAction(reduxAction);
+      onCreate?.(action);
+    },
+    [question, formSettings, handleCreateAction, onCreate],
+  );
+
+  return (
+    <FormProvider
+      initialValues={initialValues as FormValues}
+      validationSchema={ACTION_SCHEMA}
+      onSubmit={handleCreate}
+    >
+      {({ dirty }) => (
+        <Form>
+          <FormInput
+            name="name"
+            title={t`Name`}
+            placeholder={t`My new fantastic action`}
+            autoFocus
+          />
+          <FormTextArea
+            name="description"
+            title={t`Description`}
+            placeholder={t`It's optional but oh, so helpful`}
+            nullable
+          />
+          <FormModelPicker name="model_id" title={t`Model it's saved in`} />
```

Comment:
this crashes for me if I try to change it

![modelCrash](https://user-images.githubusercontent.com/30528226/212396908-28801142-89ba-49e2-8ebb-5d01e8d13a28.gif)